explorer/interpreter/interpreter.cpp

```diff
@@ -457,19 +460,20 @@ auto Interpreter::StepLvalue() -> ErrorOr<Success> {
         Address object = cast<LValue>(*val).address();
         Address field = object.ElementAddress(&access.member().member());
         return todo_.FinishAction(arena_->New<LValue>(field));
+      } else {
+        return todo_.FinishAction(act.results().back());
       }
     }
     case ExpressionKind::BaseAccessExpression: {
       const auto& access = cast<BaseAccessExpression>(exp);
       if (act.pos() == 0) {
         // Get LValue for expression.
         return todo_.Spawn(std::make_unique<LValAction>(&access.object()));
-      } else {
-        // Append `.base` element to the address, and return the new LValue.
-        Address object = cast<LValue>(*act.results()[0]).address();
-        Address base = object.ElementAddress(&access.element());
-        return todo_.FinishAction(arena_->New<LValue>(base));
       }
+      // Append `.base` element to the address, and return the new LValue.
+      Address object = cast<LValue>(*act.results()[0]).address();
+      Address base = object.ElementAddress(&access.element());
+      return todo_.FinishAction(arena_->New<LValue>(base));
```

Comment:
Might want to revert that one, except if there was a specific reason behind it?

Reply:
oh yes I will revert it.